### PR TITLE
Script to build using plain go from MAC (no bazel needed)

### PR DESCRIPTION
**Pull request by andraxylia**
**What this PR does / why we need it**:

It helps a lot when you develop from Mac and hit issues with Bazel.

Make sure you run dep ensure first to bring vendor dependencies.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by ijsnellf** [comment]
@GregHanson: you could give this a try.

File: bin/gobuild-and-push-nobazel [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        -tag) tag="$2"; shift ;;
+        -hub) hub="$2"; shift ;;
+        *) ;;
+    esac
+    shift
+done
+
+export GOOS=linux
+export GOARCH=amd64
+
```
Comment (rshriram) [inline bin/gobuild-and-push-nobazel]: Please add an echo here that tells people to run dep ensure or something..

**Comment by ldemailly** [comment]
please don't merge stuff with basic build errors:

W0906 15:22:38.218] ./bin/gobuild-and-push-nobazel:28:5: unexpected input character '$'
E0906 15:22:38.218] Build failed

please fix

**Comment by andraxylia** [comment]
Buildify for some reason thinks this is a bazel build file. I will rename the script and open an issue in the buildify repo.

**Comment by ayj** [comment]
`bin/check.sh` uses `find` to locate bazel build files and is incorrectly picking up any file that ends in `bazel` and not just the extension. See [here](https://github.com/istio/pilot/pull/1158/files#diff-2ea24c113876f7ba80e101dd5e3e77ca) for a fix.

**Comment by ldemailly** [comment]
fixed in https://github.com/istio/pilot/pull/1182

**Pull request merged**